.github/workflows/pr.yml

```diff
@@ -0,0 +1,73 @@
+name: PR
+run-name: "PR - (#${{ github.event.number }}) ${{ github.event.pull_request.title }}"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches: ['main']
+
+jobs:
+
+  format:
+    name: Verify code formatting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
```

Comment:
Shall we run on python 3.8 (as the oldest version we support) or it doesn't matter here?